### PR TITLE
Simplify representation of attribute shapes

### PR DESCRIPTION
**Pull request by axelboc**
Part of #519 

Same as #587 but with attributes.

### CODE REVIEW

File: src/h5web/providers/mock/metadata-utils.ts [hunk 1]
```diff
@@ -216,22 +214,15 @@ export function makeExternalLink(
 /* ----------------- */
 /* ----- NEXUS ----- */
 
-function makeNxStrArrayAttr(
-  array: string[],
-  attrName: 'axes' | 'auxiliary_signals'
-): HDF5Attribute {
-  return makeAttr(attrName, makeSimpleShape([array.length]), stringType, array);
-}
```
Comment (axelboc) [inline src/h5web/providers/mock/metadata-utils.ts:224]: Removing this function and calling `makeAttr` directly from the two functions below.

File: src/h5web/providers/models.ts [hunk 2]
```diff
@@ -37,10 +37,7 @@ export interface Dataset<S extends Shape = Shape, T extends HDF5Type = HDF5Type>
   type: T;
 }
 
-export type Shape = ArrayShape | ScalarShape | NullShape;
-export type ArrayShape = HDF5Dims;
-export type ScalarShape = never[];
-export type NullShape = null;
+export type NumArrayDataset = Dataset<ArrayShape, HDF5NumericType>;
```
Comment (axelboc) [inline src/h5web/providers/models.ts:40]: Just moving these things around.

File: src/h5web/providers/models.ts [hunk 3]
```diff
@@ -52,6 +49,18 @@ export interface Link<T extends HDF5Link = HDF5Link> extends Entity {
   rawLink: T;
 }
 
+export interface Attribute {
+  name: string;
+  shape: Shape;
+  type: HDF5Type;
+  value: HDF5Value;
+}
```
Comment (axelboc) [inline src/h5web/providers/models.ts:57]: New unprefixed type.

File: src/h5web/providers/hsds/utils.ts [hunk 4]
```diff
@@ -149,6 +150,13 @@ export function convertHsdsType(hsdsType: HsdsType): HDF5Type {
   }
 }
 
+export function convertHsdsAttributes(attrs: HDF5Attribute[]): Attribute[] {
+  return attrs.map((attr) => ({
+    ...attr,
+    shape: convertHsdsShape(attr.shape),
```
Comment (loichuder) [inline src/h5web/providers/hsds/utils.ts:156]: Actually, do the `type` of attribute need to be converted as well ?

The type `HDF5Attribute[]` might be wrong here.

**Review by loichuder — APPROVED** [review]

File: src/h5web/providers/hsds/utils.ts (hunk 4 above)
Comment (axelboc) [reply src/h5web/providers/hsds/utils.ts:156]: I think you're right. It probably comes from:

```ts
export type HsdsAttributeWithValueResponse = HDF5Attribute;
```

Comment (axelboc) [reply src/h5web/providers/hsds/utils.ts:156]: Fixed in my other PR. Will merge in here once you confirm it's all good: https://github.com/silx-kit/h5web/pull/589/commits/52933a8c05342e4f51613d8505b6a19c66809d76

**Pull request merged**